sdk/iot/hub/samples/README.md

```diff
@@ -56,11 +56,19 @@ For all of these samples, the following environment variables will need to be se
 - `VCPKG_ROOT`: The full path to the VCPKG directory used to generate the triplet.
 - `AZ_IOT_DEVICE_ID`: Your device id.
 - `AZ_IOT_HUB_HOSTNAME`: The hostname of your IoT Hub.
-- `AZ_IOT_DEVICE_X509_CERT_PEM_FILE`: The full path to your device cert (in `.pem` format) concatenated
- with its private key.
 - `AZ_IOT_DEVICE_X509_TRUST_PEM_FILE`: The full path to the trusted server cert (in `.pem` format). This is usually
 not needed on Mac or Linux but might be needed for Windows.
 
+For samples using certificate authentication, the following environment variable needs to be set.
+
+- `AZ_IOT_DEVICE_X509_CERT_PEM_FILE`: The full path to your device cert (in `.pem` format) concatenated
+ with its private key.
+
+For samples using SAS Key authentication, the following environment variables need to be set.
+
+- `AZ_IOT_HUB_DEVICE_SAS_KEY`: The SAS key for your device (called "Primary Key" on the Azure Portal for your device).
+- `AZ_IOT_HUB_DEVICE_SAS_KEY_EXPIRATION`: Expiration (in hours) of the SAS key. If not set, this value will default to **2 hours.**
```

Comment:
over kill?

Reply:
The expiration?

Reply:
yes, does it really need to be an env var? env var should be used for secrets that you don't want in the code that you my check in my accident

Reply:
Yea so I originally had it as just a `#define` to set in the file. But after some thinking I think it is better this way for a couple reasons:
1. Consistency with the other things that are used to get these samples working.
2. Having it as an environment variable allows the same value to be used in other samples should we implement them. Users wouldn't have to configure it for each sample.

Reply:
at the cost of more knobs to turn to get it working

Reply:
Not exactly. It has a default value of 2 hours so it'll work if the user never touches it. I think it only really adds flexibility. There isn't any additional work the user has to do.